### PR TITLE
EZP-31616: Event reference

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/EZP-31616
| Versions      | 3.x, master

### CODE REVIEW

File: docs/guide/repository/event_reference/event_reference.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class MyEventSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [
+            CopyContentEvent::class => ['onEvent', 0],
+        ];
+    }
+
+    public function onEvent(CopyContentEvent $event)
```
Comment (alongosz) [inline docs/guide/repository/event_reference/event_reference.md:29]: CS & best practices:
```suggestion
    public function onCopyContent(CopyContentEvent $event): void
```

File: docs/guide/repository/event_reference/event_reference.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+use eZ\Publish\API\Repository\Events\Content\CopyContentEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class MyEventSubscriber implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [
+            CopyContentEvent::class => ['onEvent', 0],
```
Comment (alongosz) [inline docs/guide/repository/event_reference/event_reference.md:25]: Best practices / naming: it's easier to debug an app if method name is self-commenting, e.g:
```suggestion
            CopyContentEvent::class => ['onCopyContent', 0],
```

**Review by alongosz — CHANGES_REQUESTED** [review]

**Review by alongosz — APPROVED** [review]

File: docs/guide/repository/event_reference/other_events.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+# Other events
+
+## Bookmarks
+
+The following events are dispatched adding Content items to bookmarks.
```
Comment (dabrt) [inline docs/guide/repository/event_reference/other_events.md:5]:
```suggestion
The following events are dispatched when adding Content items to bookmarks.
```

File: docs/guide/repository/event_reference/site_events.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Site events [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]
+
+The following events are dispatched when managing [Sites](../../site_factory.md):
```
Comment (dabrt) [inline docs/guide/repository/event_reference/site_events.md:3]:
```suggestion
The following events are dispatched when managing [Sites](../../site_factory.md).
```

File: docs/guide/repository/event_reference/trash_events.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Trash events
+
+The following events are dispatched when managing Trash:
```
Comment (dabrt) [inline docs/guide/repository/event_reference/trash_events.md:3]:
```suggestion
The following events are dispatched when managing Trash.
```

File: docs/guide/repository/event_reference/url_events.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+## URLs
+
+| Event | Dispatched by | Properties |
+|---|---|---|
+|`BeforeUpdateUrlEvent`|`URLService::updateUrl`|`URL $url`</br>`URLUpdateStruct $struct`</br>`URL|null $updatedUrl`|
+|`UpdateUrlEvent`|`URLService::updateUrl`|`URL $url`</br>`URLUpdateStruct $struct`</br>`URL $updatedUrl`|
+
+## URL aliases
+
+The following events are dispatched when creating and managing [URL aliases](../../url_management.md#url-aliases):
```
Comment (dabrt) [inline docs/guide/repository/event_reference/url_events.md:12]:
```suggestion
The following events are dispatched when creating and managing [URL aliases](../../url_management.md#url-aliases).
```

File: docs/guide/repository/event_reference/url_events.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+|`BeforeCreateUrlAliasEvent`|`URLAliasService::createUrlAlias`|`Location $location`</br>`private $path`</br>`private $languageCode`</br>`private $forwarding`</br>`private $alwaysAvailable`</br>`URLAlias|null $urlAlias`|
+|`CreateUrlAliasEvent`|`URLAliasService::createUrlAlias`|`Location $location`</br>`private $path`</br>`private $languageCode`</br>`private $forwarding`</br>`private $alwaysAvailable`</br>`URLAlias $urlAlias`|
+|`BeforeRefreshSystemUrlAliasesForLocationEvent`|`URLAliasService::refreshSystemUrlAliasesForLocation`|`Location $location`|
+|`RefreshSystemUrlAliasesForLocationEvent`|`URLAliasService::refreshSystemUrlAliasesForLocation`|`Location $location`|
+|`BeforeRemoveAliasesEvent`|`URLAliasService::removeAliases`|`array $aliasList`|
+|`RemoveAliasesEvent`|`URLAliasService::removeAliases`|`array $aliasList`|
+
+## URL wildcards
+
+The following events are dispatched when creating and managing [URL wildcards](../../url_management.md#url-wildcards):
```
Comment (dabrt) [inline docs/guide/repository/event_reference/url_events.md:27]:
```suggestion
The following events are dispatched when creating and managing [URL wildcards](../../url_management.md#url-wildcards).
```

File: docs/guide/repository/event_reference/user_events.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+## Assigning to User Groups
+
+| Event | Dispatched by | Properties |
+|---|---|---|
+|`BeforeAssignUserToUserGroupEvent`|`UserService::assignUserToUserGroup`|`User $user`</br>`UserGroup $userGroup`|
+|`AssignUserToUserGroupEvent`|`UserService::assignUserToUserGroup`|`User $user`</br>`UserGroup $userGroup`|
+|`BeforeUnAssignUserFromUserGroupEvent`|`UserService::unAssignUserFromUserGroup`|`User $user`</br>`UserGroup $userGroup`|
+|`UnAssignUserFromUserGroupEvent`|`UserService::unAssignUserFromUserGroup`|`User $user`</br>`UserGroup $userGroup`|
+
+## Update User data
```
Comment (dabrt) [inline docs/guide/repository/event_reference/user_events.md:34]:
```suggestion
## Updating User data
```

File: docs/guide/repository/event_reference/role_events.md [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+|---|---|---|
+|`BeforeCreateRoleDraftEvent`|`RoleService::createRoleDraft`|`Role $role`</br>`RoleDraft|null $roleDraft`|
+|`CreateRoleDraftEvent`|`RoleService::createRoleDraft`|`Role $role`</br>`RoleDraft $roleDraft`|
+|`BeforeCreateRoleEvent`|`RoleService::createRole`|`RoleCreateStruct $roleCreateStruct`</br>`RoleDraft|null $roleDraft`|
+|`CreateRoleEvent`|`RoleService::createRole`|`RoleCreateStruct $roleCreateStruct`</br>`RoleDraft $roleDraft`|
+|`BeforeUpdateRoleDraftEvent`|`RoleService::updateRoleDraft`|`RoleDraft $roleDraft`</br>`RoleUpdateStruct $roleUpdateStruct`</br>`RoleDraft|null $updatedRoleDraft`|
+|`UpdateRoleDraftEvent`|`RoleService::updateRoleDraft`|`RoleDraft $roleDraft`</br>`RoleUpdateStruct $roleUpdateStruct`</br>`RoleDraft $updatedRoleDraft`|
+|`BeforeCopyRoleEvent`|`RoleService::copyRole`|`Role $role`</br>`RoleCopyStruct $roleCopyStruct`</br>`Role|null $copiedRole`|
+|`CopyRoleEvent`|`RoleService::copyRole`|`Role $copiedRole`</br>`Role $role`</br>`RoleCopyStruct $roleCopyStruct`|
+
```
Comment (dabrt) [inline docs/guide/repository/event_reference/role_events.md:13]:
```suggestion
```

**Review by dabrt — APPROVED** [review]

**Pull request merged**